File: apps/product/rpc/internal/model/product.go
// model 包包含相关应用代码。
package model

import "time"

const (
	// StatusOffShelf 表示下架。
	StatusOffShelf int8 = 0
	// StatusOnShelf 表示上架。
	StatusOnShelf int8 = 1
)

// Product 表示商品实体（单 SKU）。
type Product struct {
	ID          int64
	SkuCode     string
	Name        string
	MainImage   string
	Description string
	PriceCent   int64
	Stock       int64
	Status      int8
	DeletedAt   *time.Time
	CreatedAt   time.Time
	UpdatedAt   time.Time
}

File: frontend/admin/src/api/modules/product.ts
import { type Int64, type Int64Like, type PaginatedList, type PaginationParams } from "@/api/core/types"
import { adminApiClient } from "@/api/core/http"

export interface AdminProduct {
  product_id: Int64
  sku_code: string
  name: string
  main_image: string
  description: string
  price_cent: number
  stock: number
  status: number
  created_at_unix: number
  updated_at_unix: number
}

export interface ProductUpsertReq {
  name: string
  main_image: string
  description: string
  price_cent: number
  stock: number
  status: number
}

export interface ListAdminProductsQuery extends PaginationParams {
  keyword?: string
  include_deleted?: boolean
}

export const adminProductApi = {
  createProduct(req: ProductUpsertReq): Promise<{ product: AdminProduct }> {
    return adminApiClient.post<{ product: AdminProduct }, ProductUpsertReq>("/products", req)
  },

  updateProduct(productId: Int64Like, req: ProductUpsertReq): Promise<{ product: AdminProduct }> {
    return adminApiClient.patch<{ product: AdminProduct }, ProductUpsertReq>(`/products/${productId}`, req)
  },

  deleteProduct(productId: Int64Like): Promise<{ product_id: Int64 }> {
    return adminApiClient.delete<{ product_id: Int64 }>(`/products/${productId}`)
  },

  getProduct(productId: Int64Like): Promise<{ product: AdminProduct }> {
    return adminApiClient.get<{ product: AdminProduct }>(`/products/${productId}`)
  },

  listProducts(query?: ListAdminProductsQuery): Promise<PaginatedList<AdminProduct>> {
    return adminApiClient.get<PaginatedList<AdminProduct>>("/products", {
      params: query,
    })
  },
}
